resolver-dns/src/main/java/io/netty/resolver/dns/UnixResolverNameServerCache.java

```diff
@@ -0,0 +1,217 @@
+                        if (i < 0) {
+                            throw new IllegalArgumentException("error parsing label " + NAMESERVER_ROW_LABEL +
+                                    " in file " + etcResolverFile + ". value: " + line);
+                        }
+                        String maybeIP = line.substring(i);
+                        // There may be a port appended onto the IP address so we attempt to extract it.
+                        if (!NetUtil.isValidIpV4Address(maybeIP) && !NetUtil.isValidIpV6Address(maybeIP)) {
+                            i = maybeIP.lastIndexOf('.');
+                            if (i + 1 >= maybeIP.length()) {
+                                throw new IllegalArgumentException("error parsing label " + NAMESERVER_ROW_LABEL +
+                                        " in file " + etcResolverFile + ". invalid IP value: " + line);
+                            }
+                            port = Integer.parseInt(maybeIP.substring(i + 1));
+                            maybeIP = maybeIP.substring(0, i);
+                        }
+                        addresses.add(new InetSocketAddress(SocketUtils.addressByName(maybeIP), port));
+                    } else if (line.startsWith(DOMAIN_ROW_LABEL)) {
+                        int i = indexOfNonWhiteSpace(line, DOMAIN_ROW_LABEL.length());
+                        if (i < 0) {
+                            throw new IllegalArgumentException("error parsing label " + DOMAIN_ROW_LABEL +
+                                    " in file " + etcResolverFile + " value: " + line);
+                        }
+                        domainName = line.substring(i);
+                        if (addresses != null && !addresses.isEmpty()) {
+                            // TODO(scott): sortlist is being ignored.
+                            // what if an entry already exists?
+                            domainToNameServerStreamMap.put(domainName, DnsServerAddresses.shuffled(addresses));
+                        }
+                        addresses = new ArrayList<InetSocketAddress>(2);
+                    } else if (line.startsWith(PORT_ROW_LABEL)) {
+                        int i = indexOfNonWhiteSpace(line, PORT_ROW_LABEL.length());
+                        if (i < 0) {
+                            throw new IllegalArgumentException("error parsing label " + PORT_ROW_LABEL +
+                                    " in file " + etcResolverFile + " value: " + line);
+                        }
+                        port = Integer.parseInt(line.substring(i));
+                    } else if (line.startsWith(SORTLIST_ROW_LABEL)) {
+                        logger.info("row type {} not supported. ignoring line: {}", SORTLIST_ROW_LABEL, line);
+                    }
+                }
+                if (addresses != null && !addresses.isEmpty()) {
+                    // TODO(scott): sortlist is being ignored.
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1135)